Packs/SplunkPy/Integrations/SplunkPy/SplunkPy.py

```diff
@@ -395,25 +403,38 @@ def fetch_notables(service: client.Service, mapper: UserMappingObject, comment_t
     # we didn't get any new incident or get less then limit
     # so the next run earliest time will be the latest_time from this iteration
     # should also set when num_of_dropped == FETCH_LIMIT
-    if not incidents or (len(incidents) + num_of_dropped) < FETCH_LIMIT:
+    if (len(incidents) + num_of_dropped) < FETCH_LIMIT:
+        demisto.debug(f'[SplunkPy] Number of fetched incidents, dropped or not, is less than {FETCH_LIMIT=}. Starting new page')
         next_run_earliest_time = latest_time
         new_last_run = {
             'time': next_run_earliest_time,
             'latest_time': None,
             'offset': 0,
             'found_incidents_ids': last_run_fetched_ids
         }
-
     # we get limit notables from splunk
     # we should fetch the entire queue with offset - so set the offset, time and latest_time for the next run
     else:
+        demisto.debug(f'[SplunkPy] Number of fetched incidents, dropped or not, is equal to {FETCH_LIMIT=}. Continue pagination')
         new_last_run = {
             'time': occured_start_time,
             'latest_time': latest_time,
             'offset': search_offset + FETCH_LIMIT,
             'found_incidents_ids': last_run_fetched_ids
         }
-    demisto.debug(f'SplunkPy - {new_last_run["time"]=}, {new_last_run["latest_time"]=}, {new_last_run["offset"]=}')
+    new_last_run['late_indexed_pagination'] = False
```

Comment:
If you look [here](https://github.com/demisto/content/blob/afef5166477ebde950c3ec751a2dc20f20ccdefb/Packs/SplunkPy/Integrations/SplunkPy/SplunkPy.py#L439), we use the **update** function, which will only update the values of the shared keys between **last_run_data** and **new_last_run**, and since **last_run_data** may have the key **late_indexed_pagination** set to True, we don't want it to stay True in the next round if not needed, therefore, I set it to be False as a default.

Reply:
is there a reason, you can see, why to use `update` instead set it with the new `last run` object?
it's look like we set the entire object each iteration

Reply:
I don't believe we need `update`, since as you said, we construct the `last_run` object every iteration

Reply:
https://github.com/demisto/content/blob/master/Packs/SplunkPy/Integrations/SplunkPy/SplunkPy.py#L393
I believe this is why we use `update`, therefore, I think we should set ** late_indexed_pagination** to be **False** as default.